Extractor/Extractor.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Bazinga\Bundle\JsTranslationBundle\Extractor;
+
+use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+use Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory;
+use Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem;
+
+abstract class Extractor extends AbstractFileExtractor implements ExtractorInterface
+{
+    const FUNCTION_STRING_ARGUMENT_REGEX = '\s?[\'"]([^"\'),]+)[\'"]\s?';
+    const REGEX_DELIMITER = '/';
+
+    private $prefix = '';
+
+    private $filesystem;
+
+    private $finderFactory;
+
+    public function __construct(Filesystem $filesystem, FinderFactory $finderFactory) {
+        $this->filesystem = $filesystem;
+        $this->finderFactory = $finderFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($resource, MessageCatalogue $catalogue)
+    {
+        $assetsPath = dirname($resource) . '/public';
```

Comment:
This extractor is meant to search for assets (which in 99% of the cases lies on the Resources/public folder). Since this checks every single file, restricting to the public folder improves the performance dramatically.